### ISSUE TITLE
Feature Request - Size of all sub directories

### ISSUE THREAD

**Issue by Zuldan** [v2kiran/PSAlphaFS#9]
I wrote a wrapper around AlphaFS a while ago. I stopped developing it when I came across PSAlphaFS. I added a feature that displayed the size of all sub directories (see below). Maybe this could be implemented as a parameter in Get-LongDirectorySize?

`Get-LongDirectorySize -Path 'C:\Program Files\' -AllSubDirectories`

```
PS C:\> Get-ChildItemEx 'C:\Program Files\' -DirectoryProperties -AllSubDirectories

    Directory: C:\Program Files

Name                           File       Directory  Size (GB)
----                           ----       ---------  ---------
7-Zip                          102        1          0
Common Files                   673        180        0.4
FileZilla FTP Client           751        100        0.02
Internet Explorer              12         3          0
Microsoft Analysis Services    23         5          0.1
Microsoft Office               4615       282        1.23
Microsoft SQL Server           1          2          0
Microsoft.NET                  1          2          0
MSBuild                        4          4          0
NVIDIA Corporation             523        23         0.65
Reference Assemblies           58         6          0.03
Windows Defender               49         1          0.01
Windows Defender Advanced      6          1          0.01
Threat Protection
Windows Mail                   9          1          0.01
Windows Media Player           42         6          0
Windows Multimedia Platform    1          0          0
Windows NT                     10         4          0.01
Windows Photo Viewer           8          1          0.01
Windows Portable Devices       1          0          0
WindowsPowerShell              133        44         0
```

**Comment by v2kiran** [comment]
this looks like a nice idea. When I made the dir size function I just did it because I found it when looking at the various classes so didn't really give it much thought.
we can implement the above but this would mean changing the current output type of 'dictionary' to pscustomobject which is okay as long as we keep the output consistent.

Do you already have the code for this, if so perhaps we can use it

**Comment by v2kiran** [comment]
Implemented in version 2.0.0.1
